src/topics/messages/subscription.rs

```diff
@@ -286,8 +277,19 @@ pub enum SubscriptionItem {
     Discontinuity(Discontinuity),
```

Comment:
we don't actually surface the discontinuities to end users in any of the other SDKs yet, do we? I thought that our stance on this was to handle them internally for now. cc: @malandis

Reply:
@kvcache voiced he wanted to know about discontinuities in his bug bash feedback.

Reply:
@kvcache we have made a deliberate choice to just make this an internal thing in all of the other SDKs so far. We can log them, and we can always expose them more formally as soon as we have a real user with a real use case. I'd like to avoid treating this SDK differently than all of the others at this point though, and we have had no customer use cases that required exposing this yet.

Reply:
I can refactor this then

Reply:
You can’t handle something like a lost message internally. That’s something the user needs a way to handle. Without exposing it in the type model, how is a user supposed to deal with missing messages?

Reply:
there is no way for them to deal with missing messages even if we expose it. This is a 2-way door, we can add it in later if someone needs it; and we may decide that there is a better way to expose it than forcing people to have a branch of their match statement that surfaces this in every place that they are consuming messages.

We don't expose discontinuities at this level in any of the other SDKs and there have not yet been any user inquiries / requests for support for it. Is there a reason we can't wait until we have a real customer use case in hand and then be a little more patient and deliberate in thinking through the options for how to expose it?

Reply:
we need to talk - I believe your opening claim is not true, and adding this branch later is a breaking change XOR you will need to make a weak match semantic on the enum that erodes the compile time safety of the sdk.

Reply:
in that case this whole thing is a bad pattern (stream with match semantic), because i find it hard to believe that we would want to commit to never adding more message types in the future. what if we want to give people a way to access heartbeats at some point? and how confident are you that we would never come up with another type of message we want to surface?

Reply:
This seems like a conversation for meatspace (or video meatspace). It seems fine to address it separately from this pr, but before you label this 1.0, if that helps you make progress more quickly?

Reply:
yeah separate pr is fine